ako-infra/avirest/handle_netinfo.go

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package avirest
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	avicache "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/cache"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/rest"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/third_party/github.com/vmware/alb-sdk/go/clients"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/third_party/github.com/vmware/alb-sdk/go/session"
+
+	"github.com/vmware/alb-sdk/go/models"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var CloudCache *models.Cloud
+
+// SyncLSLR fetches all networkinfo CR objects, compares them with the data network configured in the cloud,
+// and updates the cloud if any LS-LR data is missing.
+func SyncLSLR() {
+	lslrmap := lib.GetNetinfoCRData()
+	utils.AviLog.Infof("Got data LS LR Map: %v, from NetworkInfo CR", lslrmap)
+
+	client := avicache.SharedAVIClients().AviClient[0]
+	found, cloudModel := getAviCloudFromCache(client, utils.CloudName)
+	if !found {
+		utils.AviLog.Warnf("Failed to get Cloud data from cache")
+		return
+	}
+	matched := matchSegmentInCloud(cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs, lslrmap)
```

Comment:
This may through a `nil` pointer if the cloud is configured with `VLAN` segment I think. It is good to check if this variable cloudModel.NsxtConfiguration.DataNetworkConfig.Tier1SegmentConfig.Manual.Tier1Lrs is true in all cases. 

We could optionally just have a check where unless the cloud is of type `Overlay` we don't process anything.